src/main/java/org/folio/reader/JPathSyntaxEntityReader.java

```diff
@@ -209,16 +210,27 @@ private int getProperMatrixWidth(List<SimpleEntry<DataSource, List<ValueWrapper>
   /**
    * Applies 'readDependingOnDataSource' field to the given composite value
    */
-  private void applyReadDependingOnDataSourceFlag(CompositeValue compositeValue) {
-    compositeValue.getValue().removeIf(stringValues -> {
-      for (StringValue stringValue : stringValues) {
-        Integer dataSourceIndex = stringValue.getDataSource().getReadDependingOnDataSource();
-        if (dataSourceIndex != null) {
-          return stringValues.get(dataSourceIndex).getValue() == null;
-        }
-      }
-      return false;
-    });
+  private void applyReadIfFieldsNotEmpty(CompositeValue compositeValue) {
+    compositeValue.getValue().removeIf(stringValues -> isItemTypeValues(stringValues) && allRequiredItemFieldsAreEmpty(stringValues));
+  }
+
+  private boolean isItemTypeValues(List<StringValue> stringValues) {
+    return stringValues.stream().map(stringValue -> stringValue.getDataSource().getFrom())
+      .filter(StringUtils::isNoneEmpty)
+      .anyMatch(from -> from.contains(".items"));
+  }
+
+  /**
+   * Checks if any one item field that should be mapped in scope of a single composite value has a value.
+   * StringValue with holding hrId is not involved since always has a value.
+   *
+   * @param stringValues - item string values to verify
+   * @return true if all related item fields are empty
+   */
+  private boolean allRequiredItemFieldsAreEmpty(List<StringValue> stringValues) {
+    return stringValues.subList(0, stringValues.size() - 1).stream()
```

Comment:
Is it possible to remove subList() option?

Reply:
No, that is why I've added the javadoc